lib/puppet/type/file/content.rb

```diff
@@ -87,6 +82,10 @@ def content
     # Also, fix #872: when content is used, and replace is true, the file
     # should be insync when it exists
     def insync?(is)
+      if resource[:source] && resource[:checksum_value]
+        self.fail Puppet::Error, "Content should not exist if source and checksum_value are specified"
+      end
```

Comment:
The file type checks that you don't specify multiple ['should' values for content and source](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/type/file.rb#L366-L374). Is this check redundant?